shell.nix

```diff
@@ -1,37 +1,48 @@
+#
+# $ nix-shell --pure --argstr tag 20.09
+#
+{
+  tag ? "22.05"
+}:
 let
-    pkgs = import (builtins.fetchGit {
-      url = "https://github.com/NixOS/nixpkgs.git";
-      rev = "c2ae05d5973cc4f8842755f2807ac10e31bb2aa8";
-      ref = "master";
-    }) { };
-    pythonPackages = pkgs.python3Packages;
-    not_darwin_inputs = pkgs.lib.optionals (! pkgs.stdenv.isDarwin ) [ pythonPackages.jupyter ];
-    not_darwin_pre_shell_hook = if (! pkgs.stdenv.isDarwin) then ''
-      jupyter nbextension install --py widgetsnbextension --user
-      jupyter nbextension enable widgetsnbextension --user --py
-    '' else "";
-    filter_pyamg = builtins.filter (x: ! ((pkgs.lib.hasInfix "pyamg" x.name) && pkgs.stdenv.isDarwin));
+  pkgs = import (builtins.fetchTarball "https://github.com/NixOS/nixpkgs/archive/${tag}.tar.gz") {};
+  pypkgs = pkgs.python3Packages;
+
+  nixes_src = builtins.fetchTarball "https://github.com/wd15/nixes/archive/9a757526887dfd56c6665290b902f93c422fd6b1.zip";
```

Comment:
What is this magic tarball, and is a user account the proper source for it?

Reply:
Good catch. It's a bunch of Jupyter dependencies, [here](https://github.com/wd15/nixes/blob/master/jupyter/default.nix#L42). I don't know why I thought that was a good idea. They can just be listed directly here.

Reply:
[TODO]: check if nixpkgs has some sort of built in jupyter-complete expression with all Jupyter add ons.